apps/web/src/app/(protectedPage)/forms/[formId]/_components/formEditor/formEditorCard.tsx

```diff
@@ -138,7 +160,10 @@ export function FormEditorCard({ form }: Readonly<Props>) {
       });
       const responseJson = await response.json();
       const { fieldName } = responseJson;
-      append({ fieldName });
+      append({ fieldName, isMarkedForRemoval: false } as {
```

Comment:
You can remove `isMarkedForRemoval` now because we will use `removeFieldsIds` state for this.